reporter/sources/common.py

```diff
@@ -0,0 +1,194 @@
+"""
+Common classes allowing us to access logs from various sources
+"""
+
+import hashlib
+import logging
+
+from wikia.common.kibana import Kibana
+from wikia.common.perfmonitoring import PerfMonitoring
+
+
+class Source(object):
+    """ An abstract class for data providers to inherit from """
+    def __init__(self):
+        self._logger = logging.getLogger(self.__class__.__name__)
+
+    def query(self, query='', threshold=50):
+        """
+        The Source class entry point
+
+        - the provided query is passed to _get_entries method.
+        - returned entries are than filtered using _filter method.
+        - filtered entries are normalized using _normalize_entries method
+        - reports are grouped (using the key returned by _normalize_entries)
+        - each report is than formatted
+        """
+
+        self._logger.info("Query: '{}'".format(query))
+
+        # filter the entries
+        entries = [entry for entry in self._get_entries(query) if self._filter(entry)]
+        self._logger.info("Got {} entries after filtering".format(len(entries)))
+
+        # group them
+        normalized = self._normalize_entries(entries)
+
+        # generate reports
+        reports = self._generate_reports(normalized, threshold)
+
+        # log all reports
+        self._logger.info("Returning {} reports (with threshold set to {} applied)".format(len(reports), threshold))
+
+        for report in reports:
+            self._logger.info("> {summary} ({counter} instances)".format(
+                summary=report.get_summary(),
+                counter=report.get_counter()
+            ))
+
+        self._send_stats(query, entries, reports)
+        return reports
+
+    def _normalize_entries(self, entries):
+        """ Run all entries through _normalize method """
+        normalized = dict()
+
+        for entry in entries:
+            try:
+                key = self._normalize(entry)
+
+                # extra normalization
+                if key is not None:
+                    key = key.lower().replace(' ', '')
+            except UnicodeError:
+                # ignore UTF parsing errors
+                self._logger.error('Entry parsing error', exc_info=True)
+                continue
+
+            # all entries will be grouped
+            # using the key return by _normalize method
+            if key is not None:
+                if key not in normalized:
+                    normalized[key] = {
+                        'cnt': 0,
+                        'entry': entry
+                    }
+
+                normalized[key]['cnt'] += 1
+            else:
+                # self._logger.info('Entry not normalized: {}'.format(entry))
```

Comment:
Do you want to leave it like this? Wouldn't it be better to either remove the currently unneeded else or uncomment the log?